### ISSUE TITLE
peeling chain heuristic

### ISSUE THREAD

**Issue by rg687** [citp/BlockSci#279]
This transaction https://blockchair.com/litecoin/transaction/9b019726657fd17c166e90d66e481ab1135e6b4df4e5f89ed5a3692a4e2b3cba#i=0 seems to be a peeling chain but the heuristics finds 2 clusters. one containing the 5 previous tx and one with the four next. 

#### Reproduction Steps

```
import blocksci
peeling = blocksci.heuristics.change.peeling_chain.unique_change
cc = blocksci.cluster.ClusterManager.create_clustering(clusterdir, chain, 
                                                       should_overwrite=True, heuristic=peeling)
cm = blocksci.cluster.ClusterManager(clusterdir, chain)

addr_string = 'LWVegnkEBA79mrqNJ8sdrUYV5T6Lryq4DR'
addr = chain.address_from_string(addr_string)
print (cm.cluster_with_address(addr).index) # 4179173

addr_string = 'LhcG2skTRVgBxDTJjUngC8heg364Bseh6k'
addr = chain.address_from_string(addr_string)
print (cm.cluster_with_address(addr).index) # 4179172

```

#### System Information

BlockSci version: v0.6
Blockchain: litecoin
Parser: Disk/RPC  
Total memory: 64 GB

**Comment by maltemoeser** [comment]
The heuristic looks at the structure of the next transactions to determine whether the peeling chain is continued. In this case, both outputs are spent in a transaction that looks like they might continue the peeling chain. `unique_change` then returns no change output, since there is not a unique one

**Comment by rg687** [comment]
Ah yes thanks. Is it possible to keep all outputs that match an heuristic?

it seems that only unique_change works but not change (cant create the cluster). is it a normal behaviour?

**Comment by maltemoeser** [comment]
That's a known bug right now (#258). There's a new version of the heuristics in #259 where the normal heuristics work, but I haven't merged it into 0.6 yet since I haven't finalized the interface yet

**Comment by maltemoeser** [comment]
This should be resolved now, see my comment in #258 and the changes in #259
